clang/lib/Driver/Driver.cpp

```diff
@@ -3206,6 +3206,11 @@ void Driver::BuildInputs(const ToolChain &TC, DerivedArgList &Args,
                      InputType == types::TY_CHeader))
         Diag(clang::diag::err_drv_fsycl_with_c_type) << A->getAsString(Args);
 
+      // Emit an error if PCH(Pre-Compiled Header) file generation is forced in
+      // -fsycl mode.
+      if (IsSYCL && InputType == types::TY_CXXHeader)
+        Diag(clang::diag::err_drv_fsycl_with_pch);
```

Comment:
This is pretty generic - when we have a C++ header input, are we always guaranteed to be creating a PCH?  There may be other conditions where this may not be true and we want to be able to continue compilation.